From 9c9c744a7f7161aa45ddc2affcce68a233d5d8aa Mon Sep 17 00:00:00 2001
From: Bruno Dilly <bruno.dilly@intel.com>
Date: Tue, 12 Apr 2016 18:36:19 -0300
Subject: [PATCH 13/50] lib/sol-iio: release buffer on sol_iio_close()

Otherwise after using a Soletta application the following
issue should happen when trying to access it:

root@edison:/sys/bus/iio/devices/iio:device1# cat *raw
cat: in_voltage0_raw: Device or resource busy
cat: in_voltage1_raw: Device or resource busy
cat: in_voltage2_raw: Device or resource busy

Signed-off-by: Bruno Dilly <bruno.dilly@intel.com>
---
 src/lib/io/sol-iio.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/lib/io/sol-iio.c b/src/lib/io/sol-iio.c
index 622842c..6c674a5 100644
--- a/src/lib/io/sol-iio.c
+++ b/src/lib/io/sol-iio.c
@@ -873,6 +873,9 @@ sol_iio_close(struct sol_iio_device *device)
     }
     sol_ptr_vector_clear(&device->channels);
 
+    if ((device->buffer_enabled) && (!set_buffer_enabled(device, false)))
+        SOL_WRN("Could not disable buffer for device%d", device->device_id);
+
     if (device->fd_handler) sol_fd_del(device->fd_handler);
     if (device->fd > -1) close(device->fd);
     if (device->name_fd > -1) close(device->name_fd);
-- 
2.4.11

